src/server.cpp

```diff
@@ -55,6 +54,9 @@ void Server::handleSend(const error_code& /*error*/, std::size_t /*bytes_transfe
 
 Packet Server::makeResponse(const Packet& request)
 {
+    for (const auto& ap : request.attributes())
+        std::cout << "\t" << ap->name() << ": " << ap->value() << "\n";
```

Comment:
It would be nice to say that this list is a list of attributes. E.g.:
```
Attributes:
    USER_NAME: foobar
    FRAMED_IP_ADDRESS: 172.16.18.28
```
Not jsut a list.

Also, it is not good that you log everything in different places - packet type here, packet length in the packet constructor, request id and authenticator are not logged at all.

Let's make a special function:
```c++
void printPacket(const Packet& p);
```
You will print all information related to the packet in this function and you will call it in an appropriate place.